src/Particle/Lattice/ParticleBConds3DSoa.h

```diff
@@ -449,40 +433,37 @@ struct DTD_BConds<T, 3, PPPG + SOA_OFFSET>
 
     constexpr T minusone(-1);
     constexpr T one(1);
-#pragma omp parallel for simd aligned(temp_r, px, py, pz, dx, dy, dz)
-    for (int iat = first; iat < last; ++iat)
-    {
-      const T flip    = iat < flip_ind ? one : minusone;
-      const T displ_0 = (px[iat] - x0) * flip;
-      const T displ_1 = (py[iat] - y0) * flip;
-      const T displ_2 = (pz[iat] - z0) * flip;
 
-      const T ar_0 = -std::floor(displ_0 * g00 + displ_1 * g10 + displ_2 * g20);
-      const T ar_1 = -std::floor(displ_0 * g01 + displ_1 * g11 + displ_2 * g21);
-      const T ar_2 = -std::floor(displ_0 * g02 + displ_1 * g12 + displ_2 * g22);
+    const T flip    = iat < flip_ind ? one : minusone;
+    const T displ_0 = (px[iat] - x0) * flip;
+    const T displ_1 = (py[iat] - y0) * flip;
+    const T displ_2 = (pz[iat] - z0) * flip;
 
-      const T delx = displ_0 + ar_0 * r00 + ar_1 * r10 + ar_2 * r20;
-      const T dely = displ_1 + ar_0 * r01 + ar_1 * r11 + ar_2 * r21;
-      const T delz = displ_2 + ar_0 * r02 + ar_1 * r12 + ar_2 * r22;
+    const T ar_0 = -std::floor(displ_0 * g00 + displ_1 * g10 + displ_2 * g20);
+    const T ar_1 = -std::floor(displ_0 * g01 + displ_1 * g11 + displ_2 * g21);
+    const T ar_2 = -std::floor(displ_0 * g02 + displ_1 * g12 + displ_2 * g22);
 
-      T rmin = delx * delx + dely * dely + delz * delz;
-      int ic = 0;
-#pragma unroll(7)
-      for (int c = 1; c < 8; ++c)
-      {
-        const T x  = delx + cellx[c];
-        const T y  = dely + celly[c];
-        const T z  = delz + cellz[c];
-        const T r2 = x * x + y * y + z * z;
-        ic         = (r2 < rmin) ? c : ic;
-        rmin       = (r2 < rmin) ? r2 : rmin;
-      }
+    const T delx = displ_0 + ar_0 * r00 + ar_1 * r10 + ar_2 * r20;
+    const T dely = displ_1 + ar_0 * r01 + ar_1 * r11 + ar_2 * r21;
+    const T delz = displ_2 + ar_0 * r02 + ar_1 * r12 + ar_2 * r22;
 
-      temp_r[iat] = std::sqrt(rmin);
-      dx[iat]     = flip * (delx + cellx[ic]);
-      dy[iat]     = flip * (dely + celly[ic]);
-      dz[iat]     = flip * (delz + cellz[ic]);
+    T rmin = delx * delx + dely * dely + delz * delz;
+    int ic = 0;
+#pragma unroll(7)
```

Comment:
Is this pragma even needed these days? Was it ever needed? Compiler can see the loop size and take the appropriate decision for the architecture.

Reply:
Not sure. So should be checked when we really look at performance.